sympy/core/relational.py

```diff
@@ -925,50 +1497,882 @@ def _eval_relation(cls, lhs, rhs):
-        # We don't use the op symbol here: workaround issue #7951
-        return _sympify(lhs.__le__(rhs))
+        >>> a, b, c, d = symbols('a b c d')
+        >>> rel = Ge(a, b)
 
+        Adding a constant to both sides:
 
-Le = LessThan
+        >>> rel.add_sides(c)
+        a + c >= b + c
 
+        Adding an equality to both sides:
 
-class StrictGreaterThan(_Greater):
-    __doc__ = GreaterThan.__doc__
-    __slots__ = ()
+        >>> rel.add_sides(Eq(c, d))
+        a + c >= b + d
 
-    rel_op = '>'
+        Adding inequalities to both sides:
 
-    @classmethod
-    def _eval_relation(cls, lhs, rhs):
-        # We don't use the op symbol here: workaround issue #7951
-        return _sympify(lhs.__gt__(rhs))
+        >>> rel.add_sides(Ge(c, d))
+        a + c >= b + d
 
+        >>> rel.add_sides(Le(c, d))
+        a + d >= b + c
 
-Gt = StrictGreaterThan
+        >>> rel.add_sides(Gt(c, d))
+        a + c > b + d
 
+        >>> rel.add_sides(Lt(c, d))
+        a + d > b + c
+        """
+        return self._add_sides(arg)
 
-class StrictLessThan(_Less):
-    __doc__ = GreaterThan.__doc__
-    __slots__ = ()
+    def subtract_sides(self, arg):
+        """Subtract sides
 
-    rel_op = '<'
+        Parameters
+        ==========
 
-    @classmethod
-    def _eval_relation(cls, lhs, rhs):
-        # We don't use the op symbol here: workaround issue #7951
-        return _sympify(lhs.__lt__(rhs))
+        arg : Expr or Relational
+
+        Examples
+        ========
+
+        >>> from sympy import symbols
+        >>> from sympy.core.relational import Eq, Ne, Gt, Lt, Ge, Le
+
+        >>> a, b, c, d = symbols('a b c d')
+        >>> rel = Ge(a, b)
+
+        Subtracting a constant to both sides:
+
+        >>> rel.subtract_sides(c)
+        a - c >= b - c
+
+        Subtracting an equality to both sides:
+
+        >>> rel.subtract_sides(Eq(c, d))
+        a - c >= b - d
+
+        Subtracting inequalities to both sides:
+
+        >>> rel.subtract_sides(Ge(c, d))
+        a - d >= b - c
+
+        >>> rel.subtract_sides(Le(c, d))
+        a - c >= b - d
+
+        >>> rel.subtract_sides(Gt(c, d))
+        a - d > b - c
+
+        >>> rel.subtract_sides(Lt(c, d))
+        a - c > b - d
+        """
+        return self._subtract_sides(arg)
+
+    def multiply_sides(self, arg):
+        """Subtract sides
+
+        Parameters
+        ==========
+
+        arg : Expr or Relational
+
+        Examples
+        ========
+
+        >>> from sympy import symbols
+        >>> from sympy.core.relational import Eq, Ne, Gt, Lt, Ge, Le
+
+        >>> a, b, c, d = symbols('a b c d')
+        >>> rel = Ge(a, b)
+
+        Multiplying a constant to both sides:
+
+        >>> rel.multiply_sides(c)
+        Piecewise((a*c >= b*c, c > 0), (b*c >= a*c, c < 0), (a >= b, True))
+
+        Multiplying an equality to both sides:
+
+        >>> rel.multiply_sides(Eq(c, d))
+        Piecewise((a*c >= b*d, c > 0), (b*d >= a*c, c < 0), (a >= b, True))
```

Comment:
Does Piecewise also need these methods so `rel.multiply_sides(Eq(c, d)).multiply_sides(Eq(y,z))` works? or does Piecewise need an `apply` method that will apply a function to each of the expressions?

Reply:
Maybe ITE is better for relationals than Piecewise.

Reply:
Piecewise may simplify automatically if `c>0` is satisfied, but I may take a look at some other options